src/transpilation/bundle.ts

```diff
@@ -12,7 +12,31 @@ const createModulePath = (pathToResolve: string, program: ts.Program) =>
     escapeString(formatPathToLuaPath(trimExtension(getEmitPathRelativeToOutDir(pathToResolve, program))));
 
 // Override `require` to read from ____modules table.
-const requireOverride = `
+function requireOverride(options: CompilerOptions) {
+    if (options.luaTarget === LuaTarget.Lua50) {
+        return `
+local ____modules = {}
+local ____moduleCache = {}
+local ____originalRequire = require
+local function require(file, ...)
+    if ____moduleCache[file] then
+        return ____moduleCache[file].value
+    end
+    if ____modules[file] then
+        local module = ____modules[file]
+        ____moduleCache[file] = { value = (table.getn(arg) > 0) and module(unpack(arg)) or module(file) }
```

Comment:
Oh I didnt mean this entire string, this is just code duplication. I meant you could just store this one line that is different in a variable beforehand, then format it in later:
```ts
const setModuleCacheLine = options.luaTarget == LuaTarget.Lua50
    ? "{ value = (table.getn(arg) > 0) and module(unpack(arg)) or module(file) }"
    : "{ value = (select("#", ...) > 0) and module(...) or module(file) }"
```

Reply:
Yes, of course. Done with https://github.com/TypeScriptToLua/TypeScriptToLua/pull/1263/commits/471554074a6defbb96fb27011a9c28e508e34e34.